bids-validator/validators/json/schemas/dataset_description.json

```
@@ -52,6 +52,9 @@
     "DatasetDOI": {
       "type": "string"
     },
+    "Preregistration": {
+      "type": "string"
```

All valid strings to this field should be URIs, right?

```suggestion
      "type": "string",
      "format": "uri"
```

---

Yes ! 

But that's where I got confused because it is also the case for `DatasetDOI` is supposed to be an URI and it is just a `string` in that JSON and also in the data type column in the table in specs.

So I was trying to have things be internally consistent, because I will admit that I have no idea how the validator works and does URI validation. :see_no_evil:

---

Yes, your confusion is perfectly reasonable. The validator schema has grown over time, becoming more accurate - but some things were not improved along the way and are still like they used to be. We use https://json-schema.org/, and that's a little language in itself ... so a particular pair of contributor and reviewer may not have known that a field can get a much tighter validation than simply "string".

--> It's definitely worth it to question existing schemas and improve them if possible (improve = make them "tighter")

---

Apprently `common_definitions.json` has this:

```
"CogPOID": { "type": "string", "format": "uri" },
```

So I would assume this is it should be done in this case. Though I am now thinking if it does not make sense to have URI defined in common_definitions and then just refer to that. 

I think I need a crash course on the validator.

---

Should I add "cross check that things that are URI are encoded as such in the validator schema" on my to do list?

---

> Should I add "cross check that things that are URI are encoded as such in the validator schema" on my to do list?


@Remi-Gau that'd be very helpful 👍